mlir/docs/MIOpen/tensor_layout.md

```diff
@@ -0,0 +1,81 @@
+#Layout with group
+
+##tensorflow
+###NHWC
+<br/>
+<br/>
+####When you want to add group in tensorflow,
+####you split on dim of c or k , and tf.concat outputs 
+<br/>
+#####axis=3 split on c dim ,input layout become [N H W G C]
+input_groups = tf.split(axis=3, num_or_size_splits=groups, value=x)
+<br/>
+Example of split:
+if nhwk is 
+2,2,2,2
+[[[[1],[2]],[[3],[4]]],[[[5],[6]],[[7],[8]]]],[[[[9],[10]],[[11],[12]]],[[[13],[14]],[[15],[16]]]]
+=>
+<br/>
+after split to 2 groups :
+2,2,2,1
+2,2,2,1
+[[[[1]],[[3]]],[[[5]],[[7]]]],[[[[9]],[[11]]],[[[13]],[[15]]]]
+[[[[2]],[[4]]],[[[6]],[[8]]]],[[[[10]],[[12]]],[[[14]],[[16]]]]
+<br/>
+#####user format is [YXGCK], not final format
+weights = tf.get_variable('weights', shape=[filter_height,     
+                                                    filter_width,
+                                                    input_channels/groups,
+                                                    num_filters])
+<br/>
+#####but tensorflow will  transform filter from YXCK to KYXC:
+https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/tensorflow/core/kernels/conv_ops.cc#L1040-L1048
+<br/>
+#####axis=3 split on k dim before transform, after transform , weights layout become [GKYXC]
+weight_groups = tf.split(axis=3, num_or_size_splits=groups,value=weights)
```

Comment:
`weights = tf.get_variable(` and `weight_groups = tf.split(` need to be used together to get the TRUE group weights right? suggest to put two python statement together, or other might be confusing.

Reply:
unfortunately, it transform by tf before split, I know some guys may be confused,
but tf  don't just split  the tensor